woocommerce/woocommerce
# Add `Marketplace` class for Reactified marketplace

Related: 17315-gh-Automattic/woocommerce.com

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



- Adds Marketplace class in Internal/Admin which registers page route.
- Adds `MarketplaceServiceProvider` to make the class available.
- Adds `marketplace` item to `FeaturesController` and enables the feature by default.
- Gets the new `Marketplace` feature in `class-wc-admin-menus.php` if the feature is enabled.
- Adds `Marketplace` menu item to WooCommerce menu.
- Adds components for the `Discover` and `Extensions` tabs.

### Submission Review Guidelines:



Closes 17315-gh-Automattic/woocommerce.com. This branch is meant to provide the basic routes and pages for a new React-powered marketplace, replacing the current `Extensions` pages.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out this branch.
2. Enable the right Node version:

```
nvm use
```

3. From the WooCommerce plugin folder, update the Composer autoload class maps:

```
cd plugins/woocommerce
composer dump-autoload
```

4.  Then run the build:

```
pnpm run --filter=woocommerce build
```

5. Go to WooCommerce in your local environment, and confirm that you see the new menu item `Marketplace` in the WooCommerce menu above `Settings`. (If you're using wp-env the URL is http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fmarketplace.)
6. Check that you can navigate from there to the `Extensions` tab and back again. The `Marketplace` menu item should be highlighted as the current item while you're on each of these tabs, and the URL should correctly reflect the page and tab. If you visit one of the URLs, the correct tab should load.

### Note

Please ignore these files in the GitHub diff:

```
plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/upload-plugin.spec.js
plugins/woocommerce/changelog/e2e-fix-upload-plugin-spec
```

GitHub is highlighting changes in these files after a rebase on trunk, even though the content in this branch matches that on [trunk](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/upload-plugin.spec.js#L118) and the target branch, [feature/marketplace](https://github.com/woocommerce/woocommerce/blob/feature/marketplace/plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/upload-plugin.spec.js#L118).



### Screenshots

<img width="500" src="https://github.com/woocommerce/woocommerce/assets/1647564/518a581f-d1e7-4deb-b0d5-fb34d7dbf5c1">
